### ISSUE TITLE
Text not updated to reflect new code changes

### ISSUE THREAD

**Issue by rednewt** [RayTracing/raytracing.github.io#519]
In "Ray Tracing in One Weekend".
e.g

> I’ve often gotten into trouble using square images for debugging because I transpose x and y too often, so I’ll stick with a 200×100 image

The code right after it.
```
int main() {
    const auto aspect_ratio = 16.0 / 9.0;
    const int image_width = 384;
    const int image_height = static_cast<int>(image_width / aspect_ratio);

    std::cout << "P3\n" << image_width << " " << image_height << "\n255\n";
    point3 origin(0.0, 0.0, 0.0);
    vec3 horizontal(4.0, 0.0, 0.0);
    vec3 vertical(0.0, 2.25, 0.0);
    point3 lower_left_corner = origin - horizontal/2 - vertical/2 - vec3(0,0,1);
```

I was following the book and it was suddenly updated so I got confused, then I noticed these changes.

**Comment by rednewt** [comment]
It was 200x100 before (image_width and image_height). The text also does not explain the new code which calculates height based on aspect ratio, and the new lower left corner point calculation. It could be a source of confusion for beginners as the text contradicts the code.

**Comment by hollasch** [comment]
Thanks for reporting! I'll clean this up.

**Comment by IceTDrinker** [comment]
Hello,

I believe the two vec3s horizontal and vertical should get the aspect ratio treatment as well :)

Additionally I got to a point where the hard coded camera in Listing 27 was wrong :

```cpp
//camera.h
        camera() {
            lower_left_corner = point3(-2.0, -1.0, -1.0);
            horizontal = vec3(4.0, 0.0, 0.0);
            vertical = vec3(0.0, 2.0, 0.0);
            origin = point3(0.0, 0.0, 0.0);
        }
```

And the code code from main could just be moved to it :

```cpp
//camera.h
    camera(double aspect_ratio)
    {
        origin = point3(0.0, 0.0, 0.0);
        horizontal = vec3(4.0, 0.0, 0.0);
        vertical = vec3(0.0, horizontal.x() / aspect_ratio, 0.0);
        lower_left_corner = origin - horizontal / 2 - vertical / 2 - vec3(0, 0, 1);
    }

//main.cpp

    camera cam{aspect_ratio};
```

Otherwise you get a "distorted" image where the sphere is thinner horizontally and bigger vertically (which was noticeable on the image I produced maybe because of the high resolution I use), right before the transition to section 8 Diffuse materials.

Cheers

**Comment by hollasch** [comment]
Oh man, thanks for pointing this out. I'll prioritize this.

**Comment by rednewt** [comment]
Also I think this calculation in this lambertian material is weird. 
![image](https://user-images.githubusercontent.com/7734714/81482698-9e5d9f80-9256-11ea-973c-26dcc91be870.png)

This is how it's done in old book for reference
![image](https://user-images.githubusercontent.com/7734714/81482710-b6cdba00-9256-11ea-8ff3-600d8ac9eb13.png)

**Comment by IceTDrinker** [comment]
If you check the result the only difference is that Listing 44 uses the random_unit_vector vs random_in_unit_sphere (also there was a point vs direction approach in how the calculation was written).

And it is explained that the "old" way was a hack and that the there was a way to have a more accurate Lambertian diffusion.

From section 8.1 : 

> Really any algorithm that randomizes direction will produce surfaces that look matte. One of the simplest ways to do this turns out to be exactly correct for ideal diffuse surfaces. (I used to do it as a lazy hack that approximates mathematically ideal Lambertian.)
>
> (Reader Vassillen Chizhov proved that the lazy hack is indeed just a lazy hack and is inaccurate. The correct representation of ideal Lambertian isn't much more work, and is presented at the end of the chapter.)

Re read the section :) the code with the random_in_unit_sphere is used for "demo" purposes and then the new version is used.

If I'm mistaken let me know !

Cheers

**Comment by rednewt** [comment]
I got some help and figured it out. They removed the redundant point from equation, because it was already getting subtracted in "target - rec.p". I really don't appreciate such stealth optimizations between code snippets, it's already harder to follow and lots of explanation I believe are skipped in the book.

**Comment by IceTDrinker** [comment]
In the present case it's better to directly manipulate directions rather than points (as we are interested in the direction of the scattered ray) if anything the old code was just a case of not realizing there was no need to compute points rather than a direction.

Cheers

**Comment by hollasch** [comment]
That's unrelated to the camera code.

**Comment by rednewt** [comment]
It's not related to camera, It's a source of confusion. This is supposed to be a beginner book, so stealth adding optimization between code snippet is not a good idea.

**Comment by hollasch** [comment]
All book versions, including the original PDFs are available on our [releases page](https://github.com/RayTracing/raytracing.github.io/releases), including the three versions you prefer:
- [_Ray Tracing in One Weekend_ v1.54.0](https://github.com/RayTracing/raytracing.github.io/releases/tag/v1.54.0)
- [_Ray Tracing: The Next Week_ v1.42.0](https://github.com/RayTracing/raytracing.github.io/releases/tag/v1.42.0)
- [_Ray Tracing: The Rest of Your Life_ v1.123.0](https://github.com/RayTracing/raytracing.github.io/releases/tag/v1.123.0)

The releases above include the original associated source code.

This issue has fanned out to become a mix of multiple different issues — It's not really clear how we're supposed to schedule, prioritize or version this.

I've forked out the camera code to a focused issue: #527, scheduled for the next patch release (v3.1.1).

**Comment by trevordblack** [comment]
This should be chopped into distinct issues and dealt with. It seems like a lot of the issues brought up here have been addressed in various 3.2 PRs

**Comment by hollasch** [comment]
Original camera issue has been addressed.
